src/ImageSharp/Processing/Processors/Transforms/DegenerateTransformException.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Transforms
+{
+    /// <summary>
+    /// Represents an error that occurs during a transform operation.
+    /// </summary>
+    public sealed class DegenerateTransformException : ImageProcessingException
```

Comment:
I don't think this inheritance is a good idea. `ImageProcessingException` is usually thrown by ImageProcessor-s, the TransformBuilder concept looks fully orthogonal to image processing.

Reply:
I guess... I was thinking in processing pipelines.

Reply:
> I was thinking in processing pipelines.

Do you mean Mutate / Clone pipelines? Is it used there? In my mental model these builders are separate utilities.

Reply:
They are. I just struggle to get my head out of the pipeline way of thinking at times. I've changed it now anyway